### PR TITLE
Bluetooth: host:l2cap: fix sending error in le_ecred_reconf_req

### PR DESCRIPTION
**Pull request by KKopyscinski**
Result to send was overwritten by BT_L2CAP_RECONF_SUCCESS just after
exiting while loop. This caused to send success response even if
reconfiguration failed. Now, result is initialized to success value,
and if reconfiguration fails, this value will be overwritten with
appropriate one. Added BT_L2CAP_RECONF_INVALID_CID (0x0003) for invalid
cid in request. Added BT_L2CAP_RECONF_OTHER_UNACCEPT (0x0004) to return
if MPS is to small. Reordered checks for mps/mtu as
BT_L2CAP_RECONF_OTHER_UNACCEPT is expected to be returned if both mps
and mtu are to small.

This was affecting L2CAP/ECFC/BI-05-C and L2CAP/ECFC/BI-05-C

Signed-off-by: Krzysztof Kopyściński <krzysztof.kopyscinski@codecoup.pl>

### CODE REVIEW

File: subsys/bluetooth/host/l2cap.c [hunk 1]
```diff
@@ -1253,8 +1254,6 @@ static void le_ecred_reconf_req(struct bt_l2cap *l2cap, uint8_t ident,
 
 	BT_DBG("mtu %u mps %u", mtu, mps);
 
-	result = BT_L2CAP_RECONF_SUCCESS;
```
Comment (jhedberg) [inline subsys/bluetooth/host/l2cap.c:1256]: Removing this line and doing the initialisation upon declaration doesn't seem to be fixing anything, or be part of the fix. So please move it out to a separate patch. In fact, is it really even a good idea, since with this change the variable gets written to twice for the error cases instead of just once?

Comment (jhedberg) [reply subsys/bluetooth/host/l2cap.c:1256]: Sorry, ignore that: I should have read the commit message

**Review by jhedberg — APPROVED** [review]

**Comment by carlescufi** [comment]
@Vudentz can you please take a look?

**Review by joerchan — CHANGES_REQUESTED** [review]
Change looks OK, but could you update the commit message to use ":" instead of "/"
And preferable start with Bluetooth, so "Bluetooth: host: "

**Comment by KKopyscinski** [comment]
Updated commit message

**Review by joerchan — APPROVED** [review]

**Comment by KKopyscinski** [comment]
rebased onto main
